### PR TITLE
Improve styling of existing components in statistics popup

### PR DESCRIPTION
**Pull request by someshkhandelia**
### Summary of changes

Make tables in statistics popup look more modular

### Context and reason for change

- The existing components were not styled in a modular way.
- In future, more components can now be conveniently added to the popup.

Fix: #948

Signed-off-by: someshkhandelia <someshkhandelia@gmail.com>

### CODE REVIEW

**Comment by someshkhandelia** [comment]
Previously, the statistics popup looked like this:
image

**Comment by someshkhandelia** [comment]
After the commit https://github.com/opossum-tool/OpossumUI/pull/1006/commits/9a98a4ebb2c58032fd35f9bddcb0571b71599d81, the statistics popup looks like this:
image

File: src/Frontend/Components/ProjectStatisticsPopup/AttributionPropertyCountTable.tsx [hunk 1]
```diff
@@ -22,37 +23,44 @@ interface AttributionPropertyCountTableProps {
 export function AttributionPropertyCountTable(
   props: AttributionPropertyCountTableProps
 ): ReactElement {
+  const attributionPropertyIds: Array<string> =
+    props.attributionPropertyCountsEntries.map((entry) => entry[0].toString());
```
Comment (benedikt-richter) [inline src/Frontend/Components/ProjectStatisticsPopup/AttributionPropertyCountTable.tsx:27]: Here we could already generate the display names:

```
  const attributionPropertyDisplayNames =
    props.attributionPropertyCountsEntries.map((entry) => getAttributionPropertyDisplayNameFromId(entry[0].toString()));

```

File: src/Frontend/Components/ProjectStatisticsPopup/AttributionPropertyCountTable.tsx [hunk 2]
```diff
@@ -22,37 +23,44 @@ interface AttributionPropertyCountTableProps {
...
+  const attributionPropertyCounts: Array<string> =
```
Comment (benedikt-richter) [inline src/Frontend/Components/ProjectStatisticsPopup/AttributionPropertyCountTable.tsx:28]: Type annotation `: Array<string>` seems not necessary

File: src/Frontend/Components/ProjectStatisticsPopup/AttributionPropertyCountTable.tsx [hunk 3]
```diff
@@ -22,37 +23,44 @@ interface AttributionPropertyCountTableProps {
...
+                  key={index}
+                >
+                  {getAttributionPropertyDisplayNameFromId(
+                    attributionPropertyId
+                  )}
+                </MuiTableCell>
+              ))}
+            </MuiTableRow>
+          </MuiTableHead>
+          <MuiTableFooter style={{ position: 'sticky', bottom: 0 }}>
```
Comment (benedikt-richter) [inline src/Frontend/Components/ProjectStatisticsPopup/AttributionPropertyCountTable.tsx:50]: `style` should be replaced in favor of `sx`

File: src/Frontend/Components/ProjectStatisticsPopup/AttributionPropertyCountTable.tsx [hunk 4]
```diff
@@ -22,37 +23,44 @@ interface AttributionPropertyCountTableProps {
...
-                <MuiTableRow key={index}>
-                  <MuiTableCell
-                    sx={projectStatisticsPopupClasses.body}
-                    align="center"
-                  >
-                    {getAttributionPropertyDisplayNameFromId(
-                      attributionProperty.toString()
-                    )}
-                  </MuiTableCell>
+      <MuiTableContainer sx={{ width: '30vw', marginBottom: 3 }}>
```
Comment (benedikt-richter) [inline src/Frontend/Components/ProjectStatisticsPopup/AttributionPropertyCountTable.tsx:34]: extract `{ width: '30vw', marginBottom: 3 }`. Might be part of `projectStatisticsPopupClasses`

**Comment by benedikt-richter** [comment]
It really does look much better now 👍

File: src/Frontend/Components/ProjectStatisticsPopup/AttributionPropertyCountTable.tsx (hunk 1 above)
Comment (someshkhandelia) [reply src/Frontend/Components/ProjectStatisticsPopup/AttributionPropertyCountTable.tsx]: Should be resolved here: https://github.com/opossum-tool/OpossumUI/commit/76d8c390a41b09634d9dbc0e0f76baa1c6469d5e

File: src/Frontend/Components/ProjectStatisticsPopup/AttributionPropertyCountTable.tsx (hunk 2 above)
Comment (someshkhandelia) [reply src/Frontend/Components/ProjectStatisticsPopup/AttributionPropertyCountTable.tsx]: Should be resolved here: https://github.com/opossum-tool/OpossumUI/commit/76d8c390a41b09634d9dbc0e0f76baa1c6469d5e

File: src/Frontend/Components/ProjectStatisticsPopup/AttributionPropertyCountTable.tsx (hunk 3 above)
Comment (someshkhandelia) [reply src/Frontend/Components/ProjectStatisticsPopup/AttributionPropertyCountTable.tsx]: Should be resolved here: https://github.com/opossum-tool/OpossumUI/commit/76d8c390a41b09634d9dbc0e0f76baa1c6469d5e

File: src/Frontend/Components/ProjectStatisticsPopup/AttributionPropertyCountTable.tsx (hunk 4 above)
Comment (someshkhandelia) [reply src/Frontend/Components/ProjectStatisticsPopup/AttributionPropertyCountTable.tsx]: Should be resolved here: https://github.com/opossum-tool/OpossumUI/commit/76d8c390a41b09634d9dbc0e0f76baa1c6469d5e

**Review by benedikt-richter — CHANGES_REQUESTED** [review]
With the ORT example the table looks a bit broken.

Screenshot 2022-09-21 at 14 41 47

**Comment by someshkhandelia** [comment]
In response to: https://github.com/opossum-tool/OpossumUI/pull/1006#pullrequestreview-1115348632 , am I missing something because it looks fine on my system ?
image

**Comment by benedikt-richter** [comment]
I guess it is related to the zoom that is activated. I'll chaeck again

File: src/Frontend/Components/ProjectStatisticsPopup/shared-project-statistics-popup-styles.ts [hunk 5]
```diff
@@ -3,18 +3,34 @@
...
+    background: OpossumColors.lightBlue,
   },
   body: {
-    fontSize: 14,
+    fontSize: 11,
+    background: OpossumColors.lightestBlue,
+  },
+  attributionCountPerSourcePerLicenseTable: {
+    height: '50vh',
+    width: '40vw',
```
Comment (benedikt-richter) [inline src/Frontend/Components/ProjectStatisticsPopup/shared-project-statistics-popup-styles.ts:25]: Having this fixed width will cause the table not to show all it's contents without horizontal scrollbar

**Comment by FlorianSchepers** [comment]
> I guess it is related to the zoom that is activated. I'll check again

Yes it is zoom related. I have the same issue.

**Comment by someshkhandelia** [comment]
Learning (notes for future reference):
image
In response to https://github.com/opossum-tool/OpossumUI/pull/1006#issuecomment-1254618271, https://github.com/opossum-tool/OpossumUI/pull/1006#discussion_r977354854 and https://github.com/opossum-tool/OpossumUI/pull/1006#issuecomment-1254716089, 

- To have dynamic sizes **maxHeight** and **maxWidth** style attributes are useful. 
- Not only do they make your components dynamic to zooming but also they make it dynamic to the quantity of content contained by them. 
- Also, a point to note would be that the values must be specified in **px** units and not **vh** or **vw**, because they don't make it dynamic.

File: src/Frontend/Components/ProjectStatisticsPopup/shared-project-statistics-popup-styles.ts (hunk 5 above)
Comment (someshkhandelia) [reply src/Frontend/Components/ProjectStatisticsPopup/shared-project-statistics-popup-styles.ts]: Should be resolved here: https://github.com/opossum-tool/OpossumUI/commit/ea8ccec429cd930f657fa4a50a0e40953ecb4cb7

I verified the change with several zoom-in and zoom-out levels as well as different input files that have different content both width-wise and height-wise.

**Comment by someshkhandelia** [comment]
It was verbally agreed upon to keep the "Signal per Sources" table full-width of the popup, since in most real-life use-cases, we will be having several sources (columns).

**Comment by someshkhandelia** [comment]
- Performed rebase to main, since this PR: https://github.com/opossum-tool/OpossumUI/pull/987 was merged and it contained changes in same files as the current PR. 
 - Resolved merge conflicts that arose from the rebase.
 - Find commit : https://github.com/opossum-tool/OpossumUI/commit/1535c5d02515e4c42fa58c7e6428d41eb4717332

**Comment by someshkhandelia** [comment]
After the commit https://github.com/opossum-tool/OpossumUI/commit/1535c5d02515e4c42fa58c7e6428d41eb4717332, the popup looks like this:
image

**Comment by someshkhandelia** [comment]
It was verbally discussed with @benedikt-richter to have the column containing license names to be aligned to the left, while all the other columns must be centrally aligned

**Comment by someshkhandelia** [comment]
Incorporated https://github.com/opossum-tool/OpossumUI/pull/1006#issuecomment-1254989685. After the commit https://github.com/opossum-tool/OpossumUI/commit/6f3b2f880d8942e741bcc41787fd6459fdcb5b19, popup looks like this:
image

**Review by benedikt-richter — APPROVED** [review]

**Pull request merged**